tests/UnifiedSpecTests/Operation.php

```diff
@@ -557,6 +558,29 @@ function (IndexInfo $info) {
                     array_diff_key($args, ['to' => 1])
                 );
 
+            case 'createSearchIndex':
+                return $collection->createSearchIndex((array) $args['model']);
+
+            case 'createSearchIndexes':
+                return $collection->createSearchIndexes($args['models']);
+
+            case 'dropSearchIndex':
+                assertArrayHasKey('name', $args);
+                assertIsString($args['name']);
+
+                return $collection->dropSearchIndex($args['name']);
+
+            case 'updateSearchIndex':
+                assertArrayHasKey('name', $args);
+                assertArrayHasKey('definition', $args);
+                assertIsString($args['name']);
+                assertInstanceOf(stdClass::class, $args['definition']);
+
+                return $collection->updateSearchIndex($args['name'], $args['definition']);
+
+            case 'listSearchIndexes':
+                return $collection->listSearchIndexes($args['name'] ?? null, document_to_array($args['aggregationOptions'] ?? []));
```

Comment:
I think you'd be fine to use an `(array)` cast here, as is done for the `createDataKey` and `rewrapManyDataKey` cases in `executeForClientEncryption()`. Alternatively, you can consider  changing those to use `document_to_array()`.